examples/llm/vdb_upload/helper.py

```diff
@@ -0,0 +1,197 @@
+
+
+def setup_rss_source(pipe: Pipeline, config: Config, source_name: str, rss_config: typing.Dict[str, typing.Any]):
+    """
+    Set up the RSS source stage in the pipeline.
+
+    Parameters
+    ----------
+    pipe : Pipeline
+        The pipeline to which the RSS source stage will be added.
+    config : Config
+        Configuration object for the pipeline.
+    source_name : str
+        The name of the RSS source stage.
+    rss_config : typing.Dict[str, Any]
+        Configuration parameters for the RSS source stage.
+
+    Returns
+    -------
+    SubPipeline
+        The sub-pipeline stage created for the RSS source.
+    """
+    module_definition = RSSSourcePipeLoaderFactory.get_instance(
+        module_name=f"rss_source_pipe__{source_name}",
+        module_config={"rss_config": rss_config},
+    )
+    rss_pipe = pipe.add_stage(
+        LinearModuleSourceStage(config, module_definition, output_type=ControlMessage, output_port_name="output"))
+
+    return rss_pipe
+
+
+def setup_filesystem_source(pipe: Pipeline, config: Config, source_name: str, fs_config: typing.Dict[str, typing.Any]):
+    """
+    Set up the filesystem source stage in the pipeline.
+
+    Parameters
+    ----------
+    pipe : Pipeline
+        The pipeline to which the filesystem source stage will be added.
+    config : Config
+        Configuration object for the pipeline.
+    source_name : str
+        The name of the filesystem source stage.
+    fs_config : typing.Dict[str, Any]
+        Configuration parameters for the filesystem source stage.
+
+    Returns
+    -------
+    SubPipeline
+        The sub-pipeline stage created for the filesystem source.
+    """
+
+    module_loader = FileSourcePipeLoaderFactory.get_instance(module_name=f"file_source_pipe__{source_name}",
+                                                             module_config={"file_source_config": fs_config})
+    file_pipe = pipe.add_stage(
+        LinearModuleSourceStage(config, module_loader, output_type=ControlMessage, output_port_name="output"))
+
+    return file_pipe
+
+
+def setup_custom_source(pipe: Pipeline, config: Config, source_name: str, custom_config: typing.Dict[str, typing.Any]):
+    """
+    Setup a custom source stage in the pipeline.
+
+    Parameters
+    ----------
+    pipe : Pipeline
+        The pipeline to which the custom source stage will be added.
+    config : Config
+        Configuration object for the pipeline.
+    source_name : str
+        The name of the custom source stage.
+    custom_config : typing.Dict[str, Any]
+        Configuration parameters for the custom source stage, including
+        the module_id, module_name, namespace, and any additional parameters.
+
+    Returns
+    -------
+    SubPipeline
+        The sub-pipeline stage created for the custom source.
+    """
+
+    module_id = custom_config.pop('module_id')
+    module_name = module_id + f"__{source_name}"
```

Comment:
nit picking but I would just do
```suggestion
    module_name = f"{module_id}__{source_name}"
```